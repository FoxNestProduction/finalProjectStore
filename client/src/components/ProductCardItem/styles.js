export const sylesContainer = {
  minHeight: {
    mobile: '246px',
    tablet: '259px',
    desktop: '397px',
  },
  width: {
    mobile: '132px',
    tablet: '169px',
    desktop: '195px',
  },
  display: 'flex',
  flexDirection: 'column',
  boxShadow: '6px 71px 35px 0px rgba(229, 229, 229, 0.70)',

  '&:hover .MuiCardMedia-img': {
    transform: 'scale(1.05)',
  },
};

export const mediaBox = {
  position: 'relative',
  overflow: 'hidden',

};

export const cardMedia = {
  maxHeight: {
    mobile: '126px',
    tablet: '138px',
    desktop: '197px',
  },
  transition: '.3s',
  width: '100%',
};

export const favoriteIcon = {
  position: 'absolute',
  right: '0',
  top: '0',
  zIndex: 1050,
  borderRadius: '50%',
  bgcolor: 'rgba(255, 255, 255, 0.6)',
  '&:hover': {
    bgcolor: 'rgba(102, 79, 255, 0.3)',
  },
};

export const timeRatingBox = {
  display: 'flex',
  alignItems: 'center',
  gap: {
    mobile: '3px',
    tablet: '5px',
    desktop: '8px',
  },
};

export const priceCardBox = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
};

export const bgRatingBox = {
  display: 'flex',
  gap: '2px',
  pl: '3px',
  pr: '7px',
  alignItems: 'center',
  borderRadius: '6px',
  background: 'rgb(219, 217, 238)',
};

export const chipBox = {
  margin: {
    mobile: '2px',
    tablet: '4px',
    desktop: '8px',
  },
};

export const cartIconsButton = {
  bgcolor: 'none',
  '&:hover': {
    bgcolor: 'rgba(102, 79, 255, 0.2)',
  },
  color: 'text.primary',
};

export const cartIcons = {
  fontSize: {
    mobile: '20px',
    tablet: '25px',
    desktop: '30px',
  },
};
