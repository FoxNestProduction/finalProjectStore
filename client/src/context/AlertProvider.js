import React, { createContext, useState, useMemo, memo } from 'react';
import PropTypes from 'prop-types';

export const AlertContext = createContext({});

export const AlertContextProvider = memo(({ children }) => {
  const [alert, setAlert] = useState(false);
  const handleCloseAlert = () => {
    setAlert(false);
  };

  const handleShowAlert = () => {
    setAlert(true);
    setTimeout(() => {
      setAlert(false);
    }, 4000);
  };

  const value = useMemo(() => {
    return {
      alert,
      handleCloseAlert,
      handleShowAlert,
    };
  }, [alert]);

  return <AlertContext.Provider value={value}>{children}</AlertContext.Provider>;
});

AlertContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
