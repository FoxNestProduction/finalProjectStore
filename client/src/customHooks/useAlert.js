import { useContext } from 'react';
import AlertContext from '../context/AlertProvider';

const useAlert = () => {
  return useContext(AlertContext);
};

export default useAlert;
