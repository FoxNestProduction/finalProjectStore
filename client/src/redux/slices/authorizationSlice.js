import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  isUserAuthorized: false,
  token: null,
};

const authorizationSlice = createSlice({
  name: 'authorization',
  initialState,
  reducers: {
    setAuthorization(state, action) {
      state.isUserAuthorized = action.payload;
    },
    setToken: (state, action) => {
      state.token = action.payload;
    },
  },

});

export const {
  setAuthorization,
  setToken,
} = authorizationSlice.actions;

export default authorizationSlice.reducer;
