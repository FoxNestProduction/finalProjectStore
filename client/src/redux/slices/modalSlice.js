import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  isOpen: false,
  title: '',
  content: '',
  buttonBox: false,
  buttonAgree: {
    text: '',
    startIcon: false,
    endIcon: false,
    onClick: null,
  },
};

/* eslint-disable no-param-reassign */

const modalSlice = createSlice({
  name: 'modal',
  initialState,
  reducers: {
    openModal(state) {
      state.isOpen = true;
    },
    closeModal(state) {
      state.isOpen = false;
      state.title = '';
      state.content = '';
      state.buttonBox = false;
      state.buttonAgree.text = '';
      state.buttonAgree.startIcon = false;
      state.buttonAgree.endIcon = false;
      state.buttonAgree.onClick = null;
    },
    setTitle(state, action) {
      state.title = action.payload;
    },
    setContent(state, action) {
      state.content = action.payload;
    },
    setButtonAgree(state, action) {
      state.buttonAgree = action.payload;
    },
    addButtonBox(state) {
      state.buttonBox = true;
    },
  },
});

export const {
  openModal,
  closeModal,
  setTitle,
  setContent,
  setButtonAgree,
  addButtonBox,
} = modalSlice.actions;

/* eslint-enable no-param-reassign */

export default modalSlice.reducer;
