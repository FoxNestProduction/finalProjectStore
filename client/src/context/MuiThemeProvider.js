import React from 'react';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';

const MuiThemeProvider = ({ children }) => {
  const theme = createTheme({
    breakpoints: {
      // keys: {
      //   0: 'zero',
      //   1: 'mobile',
      //   2: 'tablet',
      //   3: 'desktop',
      //   4: 'screen',
      // },
      values: {
        zero: 0,
        mobile: 320,
        tablet: 481,
        desktop: 993,
        screen: 1200,
      },
    },
    palette: {
      primary: {
        main: '#6C5FBC',
        hover: '#664FFF',
      },
      secondary: {
        main: '#FF9140',
        hover: '#FF7C1C',
      },
      common: {
        white: '#F9F9F9',
      },
      text: {
        primary: '#2B2B2B',
        secondary: '#2B2B2B',
        header: '#606060',
        footer: '#969696',
      },
      background: {
        paper: '#F9F9F9',
        default: '#EFEFEF',
      },
      healthy: '#DAA31A',
      trending: '#FB471D',
      supreme: '#309D5B',
      bgHealthy: '#F7EDD0',
      bgTrending: '#F7C5BA',
      bgSupreme: '#33AC64',
    },
    shape: {
      borderRadius: 12,
    },
    typography: {
      fontFamily: 'Inter, sans-serif',
      body1: {
        fontFamily: 'Inter, sans-serif',
        fontWeight: 400,
      },
      body2: {
        fontFamily: 'Inter, sans-serif',
        fontWeight: 400,
      },
      button: {
        fontFamily: 'Inter, sans-serif',
        fontWeight: 500,
      },
      caption: {
        fontFamily: 'Poppins, sans-serif',
        fontWeight: 500,
      },
      h1: {
        color: '#201F1F',
        fontFamily: 'Poppins, sans-serif',
        fontWeight: 500,
      },
      h2: {
        color: '#323142',
        fontFamily: 'Poppins, sans-serif',
        fontWeight: 500,
      },
      h3: {
        color: '#2B2B2B',
        fontFamily: 'Poppins, sans-serif',
        fontWeight: 500,
      },
      subtitle1: {
        fontFamily: 'Inter, sans-serif',
      },
      subtitle2: {
        fontFamily: 'Inter, sans-serif',
      },
      // fontInter: 'Inter, sans-serif',
      // fontFamily: 'Poppins, sans-serif',
      // fontWeightSemiBold: 600,
      // fontWeightExtraBold: 800,
      // fontSize: 14,
      // fontWeightLight: 300,
      // fontWeightRegular: 400,
      // fontWeightMedium: 500,
      // fontWeightBold: 700,
    },
    button: {
      large: {
        height: '76px',
      },
      medium: {
        height: '60px',
      },
      small: {
        height: '46px',
      },
      extraSmall: {
        height: '35px',
      },
    },
    components: {
      MuiFormControl: {
        styleOverrides: {
          root: {
            // background: '#F9F9F9',
            borderRadius: '16px',
          },
        },
      },
      MuiInputBase: {
        styleOverrides: {
          root: {
            fieldset: {
              border: '1px solid #6C5FBC',
              borderRadius: '16px',
            },
            input: {
              '&::placeholder': {
                color: '#000',
                opacity: '0.8',
              },
            },
            textarea: {
              '&::placeholder': {
                color: '#000',
                opacity: '0.8',
              },
            },
          },
        },
      },
      chip: {
        width: 78,
        height: 24,
        borderRadius: 8,
        paddingTop: '10px',
        paddingLeft: '32px',
        marginRight: '10px',
      },
    },
  });

  console.log(theme);

  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
};

export default MuiThemeProvider;

MuiThemeProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
