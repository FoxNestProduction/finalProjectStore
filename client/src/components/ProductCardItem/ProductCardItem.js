import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import Box from '@mui/material/Box';
import useMediaQuery from '@mui/material/useMediaQuery';
import IconButton from '@mui/material/IconButton';
import FavoriteIcon from '@mui/icons-material/Favorite';
import AddBoxIcon from '@mui/icons-material/AddBox';
import StarRateRoundedIcon from '@mui/icons-material/StarRateRounded';
import CardContent from '@mui/material/CardContent';
import PropTypes from 'prop-types';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { useDispatch, useSelector } from 'react-redux';
import ShoppingCartCheckoutIcon from '@mui/icons-material/ShoppingCartCheckout';
import ColorChips from '../Chip/Chip';
import { chipSizeDishes } from '../Chip/styles';
import { addToCart } from '../../redux/slices/cartSlice';
import { sylesContainer, mediaBox, cardMedia, favoriteIcon, timeRatingBox, priceCardBox, bgRatingBox, chipBox, cartIcons, cartIconsButton } from './styles.js';
import { fixedEncodeURIComponent } from '../../utils/uriEncodeHelpers';

const ProductCardItem = ({ currentPrice, imageUrl, name, rating, _id }) => {
  const cart = useSelector((state) => state.cart.cart.products);
  const products = useSelector((state) => state.products.products);
  const dispatch = useDispatch();
  const [selectedItem, setSelectedItem] = useState({});
  console.log(cart);
  // console.log(selectedItem);
  const isItemInCart = cart.findIndex((product) => product.id === selectedItem.id) !== -1;
  console.log(isItemInCart);
  const handleAddToCart = (event) => {
    event.preventDefault();
    const productToCart = products.findIndex((product) => product._id === _id);
    if (productToCart !== -1) {
      const foundObject = products[productToCart];
      setSelectedItem({
        id: foundObject._id,
        currentPrice: foundObject.currentPrice,
        imageUrl: foundObject.imageUrl,
        cartQuantity: foundObject.quantity,
        name: foundObject.name,
      });
    }
    console.log(selectedItem);
    (() => dispatch(addToCart(selectedItem)))();
  };

  return (
    <Link to={`/menu/${fixedEncodeURIComponent(name)}`}>
      <Card variant="MuiCartItem" sx={sylesContainer}>
        <Box sx={mediaBox}>
          <CardActions disableSpacing sx={favoriteIcon}>
            <IconButton aria-label="add to favorites" sx={{ color: '#323142', p: '0' }}>
              <FavoriteIcon />
            </IconButton>
          </CardActions>
          <CardMedia
            component="img"
            image={imageUrl}
            title={name}
            sx={cardMedia}
          />
        </Box>
        <Box sx={chipBox}>
          <ColorChips customStyles={chipSizeDishes} />
        </Box>
        <CardContent>
          <Typography variant="h3" color="text.primary">
            {name}
          </Typography>
          <Box sx={timeRatingBox}>
            <Typography variant="body1" color="text.secondary">
              24min â€¢
            </Typography>
            <Box sx={bgRatingBox}>
              <StarRateRoundedIcon color="primary" />
              <Typography variant="body1" color="text.secondary">
                {rating}
              </Typography>
            </Box>
          </Box>
          <Box sx={priceCardBox}>
            <Typography variant="body2" color="text.primary">
              $
              {currentPrice.toFixed(0)}
              <Typography component="span" variant="body1" color="text.secondary">
                .
                {currentPrice.toFixed(2).split('.')[1]}
              </Typography>
            </Typography>
            <IconButton aria-label="add to cart" sx={cartIconsButton} onClick={handleAddToCart}>
              { isItemInCart ? (
                <AddBoxIcon sx={cartIcons} />
              ) : (
                <ShoppingCartCheckoutIcon sx={cartIcons} />
              )}
            </IconButton>
          </Box>
        </CardContent>
      </Card>
    </Link>
  );
};

ProductCardItem.propTypes = {
  currentPrice: PropTypes.number,
  imageUrl: PropTypes.string,
  name: PropTypes.string,
  rating: PropTypes.number,
  _id: PropTypes.string,
};

ProductCardItem.defaultProps = {
  currentPrice: 12.99,
  imageUrl: './img/salads/3.png',
  name: 'Chicken Hell',
  rating: 3,
  _id: '6507a306baee59670a047307',
};

export default ProductCardItem;
