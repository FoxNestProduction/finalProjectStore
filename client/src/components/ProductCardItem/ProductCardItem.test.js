import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen, } from '@testing-library/react';
import { AlertContextProvider } from '../../context/AlertProvider';
import useAlert from '../../customHooks/useAlert';
import ProductCardItem from './ProductCardItem';
import { openModal, setContent } from '../../redux/slices/modalSlice';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('../../customHooks/useAlert', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('../../redux/slices/modalSlice', () => ({
  ...jest.requireActual('../../redux/slices/modalSlice'),
  openModal: jest.fn(),
  setContent: jest.fn(),
}));

jest.mock('../../redux/slices/cartSlice', () => ({
  ...jest.requireActual('../../redux/slices/cartSlice'),
  addProductToCart: jest.fn(),
  addToCart: jest.fn(),
}));

describe('ProductCardItem Component', () => {
  const dish = {
    name: 'Product',
    description: 'About Product',
    itemNo: '123',
    currentPrice: 10.99,
    isTrending: true,
    rating: 4,
    imageUrl: 'image.jpg',
    isSupreme: false,
    isHealthy: true,
    _id: '123',
    randomNum: 25,
  };

  beforeEach(() => {
    jest.clearAllMocks();

    useAlert.mockReturnValue({
      alert: false,
      handleShowAlert: jest.fn(),
      handleCloseAlert: jest.fn(),
    });
  });

  test('should renders ProductCardItem component', () => {
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);

    const { asFragment } = render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCardItem
              name={'Product'}
              description={'About Product'}
              itemNo={'123'}
              currentPrice={10.99}
              isTrending={true}
              rating={4}
              imageUrl={'image.jpg'}
              isSupreme={true}
              isHealthy={true}
              _id={'123'}
              randomNum={36}
          />
        </AlertContextProvider>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('open Modal with LoginForm', async () => {
    const mockDispatch = jest.fn();
    require('react-redux').useDispatch.mockReturnValue(mockDispatch);
    jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);

    const { asFragment } = render(
      <MemoryRouter>
        <AlertContextProvider>
          <ProductCardItem dish={dish} />
        </AlertContextProvider>
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();

    const favouriteButton = screen.getByTestId('FavoriteBorderOutlinedIcon');

    fireEvent.click(favouriteButton);

    expect(mockDispatch).toHaveBeenCalledTimes(2);
    expect(openModal).toHaveBeenCalled();
    expect(setContent).toHaveBeenCalled();
  });

  // test('should render rating stars', async () => {
  //   const dispatch = jest.fn();
  //   mockDispatch.mockReturnValueOnce(dispatch);
  //   jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(true);
  //   jest.spyOn(require('react-redux'), 'useSelector').mockReturnValueOnce(false);

  //   render(
  //     <MemoryRouter>
  //       <AlertContextProvider>
  //         <ProductCard dish={dish} />
  //       </AlertContextProvider>
  //     </MemoryRouter>,
  //   );

  //   const ratingStar = screen.getAllByTestId('StarIcon');
  //   const rating = ratingStar.length;
  //   const expectedRating = dish.rating ?? 0;
  //   expect(rating).toBe(expectedRating);
  // });
});