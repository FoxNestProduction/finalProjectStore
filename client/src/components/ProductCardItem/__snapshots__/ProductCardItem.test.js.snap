// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardItem Component open Modal with LoginForm 1`] = `
<DocumentFragment>
  <div
    class="MuiCardActions-root MuiCardActions-spacing css-vzcucw-MuiCardActions-root"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-u1vl96-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gick0k-MuiSvgIcon-root"
        data-testid="FavoriteBorderOutlinedIcon"
        focusable="false"
        isactive="false"
        ishovered="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/menu//"
      sx="[object Object]"
    >
      <div
        class="MuiBox-root css-i6bazn"
      >
        <img
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-wlfvqy-MuiCardMedia-root"
          title=""
        />
      </div>
      <div
        class="MuiBox-root css-1oz0l0p"
      >
        <div
          class="MuiBox-root css-1li2dnr"
        />
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1i5uusm-MuiTypography-root"
      />
    </a>
    <div
      class="MuiBox-root css-fn9sqf"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-yzrrk5-MuiTypography-root"
      >
        24min
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hxfzi3-MuiSvgIcon-root"
        data-testid="FiberManualRecordIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="12"
          cy="12"
          r="8"
        />
      </svg>
      <div
        class="MuiBox-root css-130uqyg"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="StarRateRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        />
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-169s3fh-MuiTypography-root"
    >
      $null
    </p>
  </div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing css-14100l0-MuiCardActions-root"
  >
    <b>
      ADD
    </b>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ShoppingCartCheckoutIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03L21 4.96 19.25 4l-3.7 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2zM12 2l4 4-4 4-1.41-1.41L12.17 7H8V5h4.17l-1.59-1.59L12 2z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`ProductCardItem Component should renders ProductCardItem component 1`] = `
<DocumentFragment>
  <div
    class="MuiCardActions-root MuiCardActions-spacing css-vzcucw-MuiCardActions-root"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-u1vl96-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gick0k-MuiSvgIcon-root"
        data-testid="FavoriteBorderOutlinedIcon"
        focusable="false"
        isactive="false"
        ishovered="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <a
      href="/menu/product/123"
      sx="[object Object]"
    >
      <div
        class="MuiBox-root css-i6bazn"
      >
        <img
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-wlfvqy-MuiCardMedia-root"
          src="image.jpg"
          title="Product"
        />
      </div>
      <div
        class="MuiBox-root css-1oz0l0p"
      >
        <div
          class="MuiBox-root css-1ma6dsf"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1krnf8j-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Healthy
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-2mg4ts-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Trending
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1noanwr-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              Supreme
            </span>
          </div>
        </div>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1i5uusm-MuiTypography-root"
      >
        Product
      </h6>
    </a>
    <div
      class="MuiBox-root css-fn9sqf"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-yzrrk5-MuiTypography-root"
      >
        36min
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hxfzi3-MuiSvgIcon-root"
        data-testid="FiberManualRecordIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="12"
          cy="12"
          r="8"
        />
      </svg>
      <div
        class="MuiBox-root css-130uqyg"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="StarRateRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          4
        </p>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-169s3fh-MuiTypography-root"
    >
      $10.99
    </p>
  </div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing css-14100l0-MuiCardActions-root"
  >
    <b>
      ADD
    </b>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ShoppingCartCheckoutIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03L21 4.96 19.25 4l-3.7 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2zM12 2l4 4-4 4-1.41-1.41L12.17 7H8V5h4.17l-1.59-1.59L12 2z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
